content/blog/hpe-oneview-and-servicenow-integration-with-stackstorm.md

```diff
@@ -0,0 +1,180 @@
+        s = self.client
+
+        path = '/table/{0}'.format(table)
+        response = s.resource(api_path=path).create(payload=payload)
+        return response
+```
+
+Here is what the workflow looks like for the collection of alarms and sending them to ServiceNow.
+
+```yaml
+version: 1.0
+
+description: A workflow to copy HPE OneView alarms from mongo and into snow.
+
+tasks:
+  getalerts:
+    action: hpeoneview.get_mongo_alarms
+    next:
+      - when: <% succeeded() %>
+        publish:
+          - alarms: <% result().result %>
+        do: snowalerts
+
+  snowalerts:
+    with: <% ctx().alarms %>
+    action: servicenow.create_record table="u_hpeov_alarms" payload='<% item() %>'
+    next:
+      - when: <% succeeded() %>
+        do: processalarms
+
+  processalarms:
+    action: hpeoneview.process_alarms alarms=<% ctx().alarms %>
+```
+
+What if I wanted to integrate Twitter into my automation flow? Easy, **st2 pack install twitter**.  I won't show you the ServiceNow action script here but you can look at if on the exchange if you like here: [(fix-link) https://github.com/StackStorm-Exchange/stackstorm-servicenow/blob/master/actions/create_record.py](https://github.com/StackStorm-Exchange/stackstorm-servicenow/blob/master/actions/create_record.py)
+
+To finish this up, you want to set the process flag to "Yes" so you do not duplicate records into ServiceNow. It looks like the example below:
+
+```python
+import pymongo
+from lib.actions import MongoBaseAction
+
+
+class loadDb(MongoBaseAction):
+    def run(self, alarms):
+
+        mydb = self.dbclient["app_db"]
+        known = mydb["dwralarms"]
+
+        for a in alarms:
+            known.update_one({"_id":a['_id']},{"$set":{"u_process":"yes"}})
+
+        return ()
+```
+
+That's it! Once both integration packs are installed on a Stackstorm server and authorized, the rules will 'fire' every five minutes and the workflows will do the heavy lifting so you don't have to. 
+
+Finally, a diagram that shows all the moving parts of workflow "A". The rule that runs on the interval timer calls an action that, in turn, calls a workflow that calls a couple other actions. Notice that actions can be Python scripts or YAML files. It just depends on their function. 
+
+![](https://techworldwookie.com/blogpost/full-workflow.png "Workflow ")
+
+To make this a truly automated process the ServiceNow account needs to exist and the tables need to be created. In conclusion, this may seem complicated at first. In reality, its a group of small simple scripts that are linked together inside the Stackstorm framework. It also provides for the integration of many different integration packs and allows for the event based automation of many different systems. To learn more about stackstorm, you can take my tutorial https://github.com/xod442/stackstorm-tutorial and attend the StackStorm workshop on demand available here <https://hackshack.hpedev.io/workshop/21> and join the automation revolution!
```

Comment:
Suggesting hotlink again: [tutorial](https://github.com/xod442/stackstorm-tutorial) and available [here](https://hackshack.hpedev.io/workshop/21)